jackson/src/test/java/com/lightbend/lagom/serialization/JacksonJsonSerializerTest.java

```diff
@@ -211,4 +257,131 @@ public void testSerializeDone() {
+                "{'element':'a'}",
+                new TestEntityMessages.Appended("a"),
+                true),
+            new CompatData(
+                oldSerializerId,
+                "com.lightbend.lagom.serialization.BooleanCommand",
+                "{'isPublished':false}",
+                BooleanCommand.builder().isPublished(false).build(),
+                true),
+            new CompatData(
+                oldSerializerId,
+                "com.lightbend.lagom.serialization.BooleanCommand",
+                "{'isPublished':true}",
+                BooleanCommand.builder().isPublished(true).build(),
+                true),
+            new CompatData(
+                oldSerializerId,
+                "com.lightbend.lagom.serialization.OptionalCommand",
+                "{'name':'Bob','maybe':'SomeOrg'}",
+                OptionalCommand.builder().name("Bob").maybe(Optional.of("SomeOrg")).build(),
+                true),
+            new CompatData(
+                oldSerializerId,
+                "com.lightbend.lagom.serialization.OptionalCommand",
+                "{'name':'Bob','maybe':null}",
+                OptionalCommand.builder().name("Bob").build(),
+                true),
+            new CompatData(
+                oldSerializerId,
+                "com.lightbend.lagom.serialization.OptionalCommand",
+                "{'name':'Bob'}",
+                OptionalCommand.builder().name("Bob").build(),
+                false), // leaving out the optional maybe field
+            new CompatData(
+                oldSerializerId,
+                "com.lightbend.lagom.serialization.TestEntityMessages$ChangeMode",
+                "{'mode':'PREPEND'}",
+                new TestEntityMessages.ChangeMode(TestEntityMessages.Mode.PREPEND),
+                true),
+            new CompatData(
+                oldSerializerId,
+                "com.lightbend.lagom.serialization.TestEntityMessages$Get",
+                "{}",
+                TestEntityMessages.Get.instance(),
+                true),
+            new CompatData(
+                oldSerializerId,
+                "com.lightbend.lagom.serialization.TestEntityMessages$InPrependMode",
+                "{}",
+                TestEntityMessages.InPrependMode.instance(),
+                true),
+            new CompatData(
+                oldSerializerId,
+                "com.lightbend.lagom.serialization.PostContent",
+                "{'title':'Some Title','body':'Content...','comments':[{'author':'user1','content':'+1'},{'author':'user2','content':'-1'}]}",
+                content,
+                true),
+            new CompatData(
+                oldSerializerId,
+                "com.lightbend.lagom.serialization.Greeting",
+                "{'message':'Hello','timestamp':[2019,5,23,9,54,32,174000000]}",
+                Greeting.of("Hello", LocalDateTime.of(2019, 5, 23, 9, 54, 32, 174000000)),
```

Comment:
`LocalDateTime` as an array is not a great default by Jackson IMHO. It probably is faster than ISO8601 but 🤷🏼‍♂️(same goes for all [java 8 date/time](https://github.com/FasterXML/jackson-modules-java8/tree/master/datetime) defaults in Jackson).

WRT this test, we should add a check for an instance of `java.time.Instant` too.

I think these Jackson defaults is another reason to have two `ObjectMapper` instances in Lagom: one for external API using ISO8601 (and other defaults) and another for internal use (cluster comms, journaled events,...) with performance/size/... in mind. Again, having separate `ObjectMapper` instances should be solved on a separate PR.

Reply:
I have changed the default in https://github.com/akka/akka/pull/27056 for better interoperability. Yes, it is slower.

Good news is that it seem like deserialization of both formats are supported so it can be changed in Lagom without impact to reading old data or rolling update.

Reply:
> Good news is that it seem like deserialization of both formats are supported so it can be changed in Lagom without impact to reading old data or rolling update.

Wow! Ok, this is really useful for internal messages and rolling upgrades. We'll have to consider API messages though. Still, really nice find.

Reply:
For the record, I tested it here
https://github.com/akka/akka/pull/27056/files#diff-e26f56a36a03b3d287e0e0bd444fbca4R160
and here https://github.com/akka/akka/pull/27056/files#diff-e26f56a36a03b3d287e0e0bd444fbca4R186

Reply:
Default format changed in Akka 2.6.0-M3

Reply:
> Default format changed in Akka 2.6.0-M3

We'll have to document how to create an `ObjectMapper` for the default `MessageSerializer` in the ServiceAPI so users can opt into the old date format.

Reply:
We can add this on `manual/java/releases/Migration16.md`: 

```
The default settings for the `ObjectMapper` in Lagom now use `ISO-8601` date formats. The default in previous versions of Lagom was to use Jackson private format which can be more time and/or space efficient. If you want your Service API to produce the same output for types like `java.time.Instant`or `java.time.LocalDateTime` tune the `ObjectMapper`. In your `application.conf` use for example:

akka.serialization.jackson {
  # Configuration of the ObjectMapper for external service api
  jackson-json-serviceapi {
     WRITE_DATES_AS_TIMESTAMPS = on # Serializes dates using Jackson custom formats
  }
}
```

Reply:
yes, that looks correct